homeassistant/components/roomba/__init__.py

```diff
@@ -169,6 +170,9 @@ def _has_all_unique_bilds(value):
     )
     if unload_ok:
         domain_data = hass.data[DOMAIN][config_entry.entry_id]
+        # unsubscribe listerner
+        domain_data[LISTENER]()
+        # disconnect vacuum
         await async_disconnect_or_timeout(hass, roomba=domain_data[ROOMBA_SESSION])
```

Comment:
Nothing is checking the return value of `async_disconnect_or_timeout`.

Reply:
I added a control loop for disconnection because the API does not return status

Reply:
I was commenting about the return value of `async_disconnect_or_timeout`. We don't check that anywhere. Either we should check it or remove the explicit return value.

Reply:
Returning True somewhere means there should be a case where we return False.

Reply:
As I just added the disconnection control, do you see a drawback to the fact that I add the return on the function "await async_disconnect_or_timeout" which will return True if it is successfully played
And so I go out with a return to False at the end.

```
>     if unload_ok:
>         domain_data = hass.data[DOMAIN].pop(config_entry.entry_id)
>         # unsubscribe listerner
>         domain_data[LISTENER]()
>         # disconnect vacuum
>         return await async_disconnect_or_timeout(hass, roomba=domain_data[ROOMBA_SESSION])
>     return False
```

Reply:
We're not catching the raised exception here from `async_disconnect_or_timeout`. And that function still only returns `True` never `False`.